source/simulator/helper_functions.cc

```diff
@@ -909,8 +909,142 @@ namespace aspect
+    FEValues<dim> fe_values_xy (mapping,
+                                finite_element,
+                                quadrature_formula_xy,
+                                update_values   |
+                                update_quadrature_points |
+                                update_JxW_values);
+    std::vector<double> compositional_values_xy(n_q_points_xy);
+    FEValues<dim> fe_values_yx (mapping,
+                                finite_element,
+                                quadrature_formula_yx,
+                                update_values   |
+                                update_quadrature_points |
+                                update_JxW_values);
+    std::vector<double> compositional_values_yx(n_q_points_yx);
+
+    std::vector<types::global_dof_index> local_dof_indices (finite_element.dofs_per_cell);
+
+    typename DoFHandler<dim>::active_cell_iterator
+    cell = dof_handler.begin_active(),
+    endc = dof_handler.end();
+    double local_area_integrals;
+    std::vector<double> local_compositional_integrals (parameters.n_compositional_fields);
+    std::vector<double> local_compositional_average_integrals (parameters.n_compositional_fields);
+    const double C_max=1.0;//FIXME need to add into the parameters list
+    const double C_min=0;
+    double min_composition=1.0;
+    double max_composition=0;
+
+#define MPI_DG
+
+#ifdef MPI_DG
+    LinearAlgebra::BlockVector distributed_vector (introspection.index_sets.system_partitioning,
+                                                   mpi_communicator);
+    distributed_vector = vector;
+#else
+    LinearAlgebra::BlockVector distributed_vector(vector);
+#endif
+
+    for (; cell != endc; ++cell)
+      {
+        if (cell->is_locally_owned())
+          {
+            cell->get_dof_indices (local_dof_indices);
+            fe_values_xy.reinit (cell);
+            fe_values_yx.reinit (cell);
+
+            local_area_integrals=0;
+
+            for (unsigned int q=0; q<n_q_points_xy; ++q)
+              {
+                local_area_integrals+= fe_values_xy.JxW(q);
+              }
+            for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
+              {
+                fe_values_xy[introspection.extractors.compositional_fields[c]].get_function_values(vector,
+                    compositional_values_xy);
+                fe_values_yx[introspection.extractors.compositional_fields[c]].get_function_values(vector,
+                    compositional_values_yx);
+
+                local_compositional_average_integrals[c]=0;
+
+                for (unsigned int q=0; q<n_q_points_xy; ++q)
+                  {
+                    local_compositional_average_integrals[c]+= compositional_values_xy[q]*fe_values_xy.JxW(q);
+                  }
+                local_compositional_average_integrals[c] /= local_area_integrals;
+
+                min_composition=compositional_values_xy[0];
+                max_composition=compositional_values_xy[0];
```

Comment:
declaring min_composition and max_composition right here would probably be not much slower (its just a number), but would save a line of code for each variable. It would also limit the scope of the variables, which is always a good thing to do.